service/src/test/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiControllerGcsBucketTest.java

```diff
@@ -116,35 +115,81 @@ public class ControlledGcpResourceApiControllerGcsBucketTest extends BaseConnect
   private UUID workspaceId2;
   private String projectId2;
 
-  private String sourceResourceName = TestUtils.appendRandomNumber("source-resource-name");
-  private String sourceBucketName = TestUtils.appendRandomNumber("source-bucket-name");
+  private final String sourceResourceName = TestUtils.appendRandomNumber("source-resource-name");
+  private final String sourceBucketName = TestUtils.appendRandomNumber("source-bucket-name");
   private ApiGcpGcsBucketResource sourceBucket;
 
+  private final String source2ResourceName = TestUtils.appendRandomNumber("source-resource-name");
+  private final String source2BucketName = TestUtils.appendRandomNumber("source-bucket-name");
+  private ApiGcpGcsBucketResource source2Bucket;
+
   // See here for how to skip workspace creation for local runs:
   // https://github.com/DataBiosphere/terra-workspace-manager#for-local-runs-skip-workspacecontext-creation
   @BeforeAll
   public void setup() throws Exception {
-    workspaceId =
-        mockMvcUtils
-            .createWorkspaceWithCloudContext(userAccessUtils.defaultUserAuthRequest())
-            .getId();
+    // Make sure no debug info is set
+    jobService.setFlightDebugInfoForTest(null);
+
+    // To work around the permission propagation delay, I removed the per-test
+    // grant and revoke calls. Instead, we use the following fixed setup:
+    //
+    // defaultUser - owner of both workspaces
+    // secondUser - writer on workspace1, reader on workspace2
+    // noBillingUser - no access on both workspaces
+    //
+    // We split the create workspace from the cloud context so that we can
+    // do the grants and get the Sam group set before we sync the permissions
+    // to the project. The owner gets a temporary grant, but we need to wait
+    // for propagation of the other permissions. So the
+    // sequence is:
+    //
+    // 1. create workspaces as default user
+    // 2. grant permissions to 2nd user
```

Comment:
Sam does not create the Google groups until we do the sync as part of context create. So by doing the grant before that point, we avoid a GCP IAM propagation. What was there before did:
1. create context and propagate the Google groups with the owner in them
2. grant second user updates the the Google group and makes a second GCP IAM propagation.
Since GCP has just done a propagation, it has the answer cached and it take a very long time for the propagation to happen.

I still need the waitForProjectAccess since the second user does not get the benefit of the temporary grant mechanism.